Build failure: obs-studio-plugins.obs-ndi

## Steps To Reproduce

Steps to reproduce the behavior:

1. Build `obs-studio-plugins.obs-ndi` with `NIXPKGS_ENABLE_UNFREE=1` on master
2. See errors in `buildPhase`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/5nwl9lb937f4fc0va65az5lgpsnqjk3d-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/4vskdbvdhmqj1mxmrnhb2mr706sqi1h0-hardcode-ndi-path.patch
patching file src/plugin-main.cpp
Hunk #1 succeeded at 304 (offset 60 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/share/doc/obs-ndi -DCMAKE_INSTALL_INFODIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/2r2xi5pbg29bsmqywsm5zgl8l7adky4i-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/2r2xi5pbg29bsmqywsm5zgl8l7adky4i-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/2r2xi5pbg29bsmqywsm5zgl8l7adky4i-gcc-wrapper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/p6l7pq6r4asr4wajmrn2pvb8048npq50-obs-ndi-4.14.1 -DENABLE_QT=ON
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/2r2xi5pbg29bsmqywsm5zgl8l7adky4i-gcc-wrapper-13.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/2r2xi5pbg29bsmqywsm5zgl8l7adky4i-gcc-wrapper-13.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Threads: TRUE
-- Found CURL: /nix/store/ygwnjglxwvmladk25fsdghzh2cacczfg-curl-8.11.0/lib/libcurl.so (found version "8.11.0")
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/1kpzmzzbr72ppjml5jdvc1dk4n23dvfj-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /nix/store/ack6ad8jjj8g10d7mi46vr401gkr239q-libxkbcommon-1.7.0/lib/libxkbcommon.so (Required is at least version "0.5.0")
-- Found WrapVulkanHeaders: /nix/store/id5x3px8f4cilyxb4php4q0dlp04h5vs-vulkan-headers-1.3.296.0/include
-- Configuring done (2.8s)
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j80 SHELL=/nix/store/4vvkmiq95x6zzz82lcsbq8607x5v7aws-bash-5.2p37/bin/bash
[  5%] [32mBuilding C object CMakeFiles/plugin-support.dir/plugin-support.c.o[0m
[ 11%] [32m[1mLinking C static library libplugin-support.a[0m
[ 11%] Built target plugin-support
[ 11%] Built target obs-ndi_autogen_timestamp_deps
[ 17%] [34m[1mAutomatic MOC and UIC for target obs-ndi[0m
[ 17%] Built target obs-ndi_autogen
[ 23%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/obs-ndi_autogen/mocs_compilation.cpp.o[0m
[ 41%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/obs-support/shared-update.cpp.o[0m
[ 41%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/forms/obsndi-update.cpp.o[0m
[ 41%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/forms/output-settings.cpp.o[0m
[ 47%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/obs-support/remote-text.cpp.o[0m
[ 52%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/main-output.cpp.o[0m
[ 58%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/obs-ndi-filter.cpp.o[0m
[ 64%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/plugin-main.cpp.o[0m
[ 70%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/Config.cpp.o[0m
[ 82%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/obs-ndi-output.cpp.o[0m
[ 82%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/obs-ndi-source.cpp.o[0m
[ 88%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/premultiplied-alpha-filter.cpp.o[0m
[ 94%] [32mBuilding CXX object CMakeFiles/obs-ndi.dir/src/preview-output.cpp.o[0m
[01m[K/build/source/src/forms/obsndi-update.cpp:[m[K In constructor '[01m[KObsNdiUpdate::[01;32m[KObsNdiUpdate[m[K(const PluginUpdateInfo&, QWidget*)[m[K':
[01m[K/build/source/src/forms/obsndi-update.cpp:177:40:[m[K [01;31m[Kerror: [m[K'[01m[Kvoid QDateTime::[01;32m[KsetTimeSpec[m[K(Qt::TimeSpec)[m[K' is deprecated: Use setTimeZone() instead [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations]8;;[m[K]
  177 |                 [01;31m[KutcDateTime.setTimeSpec(Qt::UTC)[m[K;
      |                 [01;31m[K~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~[m[K
In file included from [01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtCore/qcborvalue.h:10[m[K,
                 from [01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtCore/qjsonvalue.h:7[m[K,
                 from [01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtCore/qjsondocument.h:8[m[K,
                 from [01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtCore/QJsonDocument:1[m[K,
                 from [01m[K/build/source/src/forms/obsndi-update.h:20[m[K,
                 from [01m[K/build/source/src/forms/obsndi-update.cpp:32[m[K:
[01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtCore/qdatetime.h:404:10:[m[K [01;36m[Knote: [m[Kdeclared here
  404 |     void [01;36m[KsetTimeSpec[m[K(Qt::TimeSpec spec);
      |          [01;36m[K^~~~~~~~~~~[m[K
[01m[K/build/source/src/forms/obsndi-update.cpp:193:37:[m[K [01;31m[Kerror: [m[K'[01m[Kvoid QCheckBox::[01;32m[KstateChanged[m[K(int)[m[K' is deprecated: Use checkStateChanged() instead [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations]8;;[m[K]
  193 |                         &QCheckBox::[01;31m[KstateChanged[m[K, this, [](int state) {
      |                                     [01;31m[K^~~~~~~~~~~~[m[K
In file included from [01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtWidgets/QCheckBox:1[m[K,
                 from [01m[K/build/source/build/obs-ndi_autogen/include/ui_obsndi-update.h:14[m[K,
                 from [01m[K/build/source/src/forms/obsndi-update.h:24[m[K:
[01m[K/nix/store/ic304qmqb3fsld8s3w25vikbq6m12ivl-qtbase-6.8.0/include/QtWidgets/qcheckbox.h:41:10:[m[K [01;36m[Knote: [m[Kdeclared here
   41 |     void [01;36m[KstateChanged[m[K(int);
      |          [01;36m[K^~~~~~~~~~~~[m[K
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/obs-ndi.dir/build.make:90: CMakeFiles/obs-ndi.dir/src/forms/obsndi-update.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:115: CMakeFiles/obs-ndi.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.163, NixOS, 24.11 (Vicuna), 24.11.20240723.2874fc4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/home/tomodachi94/nixpkgs` (checked out to `1cbb18a94ad1c9066343b098edcbfd165f5fad1d`)

## Notify maintainers

@jshcmpbll

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/obs-studio/plugins/obs-ndi/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/obs-ndi/default.nix
@@ -47,6 +47,8 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [ "-DENABLE_QT=ON" ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-deprecated-declarations";
+
   dontWrapQtApps = true;
 
   meta = with lib; {

